### ISSUE TITLE
Library doesn't allow other arguments to be in the redirect-URI

### ISSUE THREAD

**Issue by dsoprea** [zmb3/spotify#19]
I'm assigning a redirect-URI like:

```
redirectUrl := "http://urlecho.appspot.com/echo?debugMode=1"
auth := spotify.NewAuthenticator(redirectUrl, spotify.ScopeUserReadPrivate)
```

..but the produced authorization URL looks like:

```
https://accounts.spotify.com/authorize?client_id=CLIENTID&redirect_uri=http%3A%2F%2Furlecho.appspot.com%2Fecho&response_type=code&scope=user-read-private&state=arbitrary-state-data
```

When I add the argument to the URL directly, Spotify accepts it (though I have to make sure to white-list the whole thing, including the query portion) and then redirects to it. So, it is definitely supported.

**Comment by zmb3** [comment]
Please make sure you update all dependencies with `go get -u github.com/zmb3/spotify`.

This seems to work for me.

```
❯ cat main.go
package main

import (
	"fmt"

	"github.com/zmb3/spotify"
)

func main() {
	redirect := "http://urlecho.appspot.com/echo?debugMode=1"
	auth := spotify.NewAuthenticator(redirect, spotify.ScopeUserReadPrivate)
        auth.SetAuthInfo("CLIENT_ID", "CLIENT_SECRET")
	authURL := auth.AuthURL("state")
	fmt.Println("Auth URL:", authURL)
}

❯ go run main.go
Auth URL: https://accounts.spotify.com/authorize?client_id=CLIENT_ID&redirect_uri=http%3A%2F%2Furlecho.appspot.com%2Fecho%3FdebugMode%3D1&response_type=code&scope=user-read-private&state=state
```

**Comment by dsoprea** [comment]
Weird phenomenon:

1. I rebuilt and re-ran to again confirm that the bug happened as I claimed it did. It did.
2. I did a *git pull* to bring down updates. I was already up-to-date.
3. I killed the project in my *GOPATH*, did a *go get*, rebuilt, and re-ran. It now works as it should've.

I cloned the repository last week, the last commit was on the 1st of December, and it wasn't even relevant. So, I'm not clear what changed.
